templates/support/SecurityClient.swift.hbs

```diff
@@ -0,0 +1,361 @@
+//  
+//  {{>frag/generatedBy}}
+//
+
+import Foundation
+
+public protocol SecurityClient {
+    /** An API request received a 401 response */
+    func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws
+    
+    /** An authorization header to the request */
```

Comment:
```swift
/** Modify the request to authorise it for the required SecurityScheme and scopes. */
```

Perhaps we should move the `request` parameter to the first position? That might _read_ better.